### PR TITLE
fix(ComposedModal): fix button variants

### LINKED ISSUE #3326: ComposedModal Component Styling
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

When setting the danger property to true on the Modal component, the left button’s kind property is set to secondary (black) while the button on the right has its kind property set to danger (red). See the first image. The button on the right is the same in the ComposedModal component, but the button on the left is set based on this conditional statement in the component’s source code:

`kind={danger ? 'tertiary' : 'secondary'}`

When I set the danger property to true on the ComposedModal, it looks like the second image attached. The button on the left is gray instead of black like it is in the Modal example. It seems like that the button’s kind property is being assigned to tertiary in ComposedModal. Is this supposed to happen? I was told the button on the left should be black.

DangerModalExample
DangerComposedModalExample

> Is this issue related to a specific component?

ComposedModal

> What did you expect to happen? What happened instead? What would you like to
> see changed?

I expected the cancel button in the second image to be black instead of gray by default. I can set the kind property myself to change the colors, but I just want to be sure this is intended behavior.

> What browser are you working in?

Google Chrome

> What version of the Carbon Design System are you using?

v10

## Steps to reproduce the issue

1. Set danger property of ComposedModal and ModalFooter to true
2. Observe button colors

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

I cannot get the sandbox to recognize ComposedModal. See images attached.

### PR DESCRIPTION
**Pull request by asudoh**
Fixes #3326.

#### Changelog

**Changed**

- Buttons in `<ComposedModal>`, aligning to `<Modal>` ones.

#### Testing / Reviewing

Testing should make sure `<ComposedModal>` is not broken.

### CODE REVIEW

File: packages/react/src/components/ComposedModal/ComposedModal-story.js [hunk 1]
```diff
@@ -98,12 +98,8 @@ storiesOf('ComposedModal', module)
           </p>
         </ModalBody>
         <ModalFooter>
-          <Button
-            kind={props.composedModal().danger ? 'tertiary' : 'secondary'}>
-            Cancel
-          </Button>
-          <Button
-            kind={props.composedModal().danger ? 'danger--primary' : 'primary'}>
+          <Button kind={'secondary'}>Cancel</Button>
```
Comment (emyarod) [inline packages/react/src/components/ComposedModal/ComposedModal-story.js]: we can just do this right?

```suggestion
          <Button kind="secondary">Cancel</Button>
```

**Review by emyarod — APPROVED** [review]
looks good to me!

File: packages/react/src/components/ComposedModal/ComposedModal-test.js [hunk 2]
```diff
@@ -146,10 +146,10 @@ describe('<ModalFooter />', () => {
       expect(buttonComponent.props().kind).toBe('danger');
     });
 
-    it('renders tertiary button if secondary text && danger', () => {
+    it('renders secondary button if secondary text && danger', () => {
       const buttonComponent = secondaryWrapper.find(Button);
       expect(buttonComponent.exists()).toBe(true);
-      expect(buttonComponent.props().kind).toBe('tertiary');
+      expect(buttonComponent.props().kind).toBe('secondary');
```
Comment (joshblack) [inline packages/react/src/components/ComposedModal/ComposedModal-test.js]: An alternative:

```suggestion
      expect(buttonComponent.prop('kind')).toBe('secondary');
```

File: packages/react/src/components/ComposedModal/ComposedModal.js [hunk 3]
```diff
@@ -493,7 +493,7 @@ export class ModalFooter extends Component {
           <Button
             className={secondaryClass}
             onClick={this.handleRequestClose}
-            kind={danger ? 'tertiary' : 'secondary'}>
+            kind={'secondary'}>
```
Comment (joshblack) [inline packages/react/src/components/ComposedModal/ComposedModal.js]:
```suggestion
            kind="secondary">
```

**Review by joshblack — CHANGES_REQUESTED** [review]
Left some minor notes!

**Review by jeanservaas — APPROVED** [review]
LGTM

**Comment by asudoh** [comment]
Good catch @emyarod and @joshblack - Fixed.

**Review by joshblack — APPROVED** [review]

**Pull request merged**